__tests__/app.test.js

```diff
@@ -29,16 +29,41 @@ describe("app", () => {
           .get("/api/topics/")
           .expect(200)
           .then(({ body }) => {
-            expect(body).toEqual([
-              { slug: "mitch", description: "The man, the Mitch, the legend" },
-              { slug: "cats", description: "Not dogs" },
-              { slug: "paper", description: "what books are made of" },
-            ]);
+            expect(typeof body).toBe("object");
             body.forEach((topic) => {
               expect(Object.keys(topic)).toEqual(["slug", "description"]);
             });
           });
       });
     });
+    describe("GET /api/articles/:article_id", () => {
+      test("Status 200: Should respond with an article object using the id", () => {
+        return request(app)
+          .get("/api/articles/1")
+          .expect(200)
+          .then((res) => {
+            expect(typeof res.body).toBe("object");
+          });
+      });
+      test("Status 400: Should respond with a an error message when given invalid id", () => {
+        return request(app)
+          .get("/api/articles/not-an-article")
+          .expect(400)
+          .then((res) => {
+            expect(res.body.msg).toBe("Bad request");
+          });
+      });
+      test("Status 404: Should respond with a an error message when given a valid but not existing id", () => {
+        article_id = 999;
+        return request(app)
+          .get(`/api/articles/${article_id}`)
+          .expect(404)
+          .then((res) => {
+            expect(res.body.msg).toBe(
+              `No article found for article: ${article_id}`
+            );
+          });
+      });
+    });
   });
```

Comment:
Good work implementing the testing for an invalid id, and a valid id with no entry. The status codes are correct and an appropriate response message is being returned to the client, well done.